x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/field_config.tsx

```diff
@@ -1251,4 +1281,168 @@ export const FIELD = (readOnly?: boolean): FieldMap => ({
+        values={{
+          all: <EuiCode>all</EuiCode>,
+          any: <EuiCode>any</EuiCode>,
+        }}
+      />
+    ),
+    props: (): EuiSelectProps => ({
+      options: Object.values(Mode).map((value) => ({
+        value,
+        text: value,
+      })),
+      disabled: readOnly,
+    }),
+  },
+  [ConfigKey.RESPONSE_BODY_MAX_BYTES]: {
+    fieldKey: ConfigKey.RESPONSE_BODY_MAX_BYTES,
+    component: FieldNumber,
+    label: i18n.translate('xpack.synthetics.monitorConfig.responseBodyMaxBytes.label', {
+      defaultMessage: 'Response body max bytes',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.responseBodyMaxBytes.helpText', {
+      defaultMessage: 'Control the maximum size of the stored body contents.',
+    }),
+    hidden: (dependencies) => {
+      const [responseBodyIndex] = dependencies || [];
+      return responseBodyIndex === ResponseBodyIndexPolicy.NEVER;
+    },
+    props: (): EuiFieldNumberProps => ({ min: 1, step: 'any', readOnly }),
+    dependencies: [ConfigKey.RESPONSE_BODY_INDEX],
+  },
+  [ConfigKey.IPV4]: {
+    fieldKey: ConfigKey.IPV4, // also controls ipv6
+    component: ComboBox,
+    label: i18n.translate('xpack.synthetics.monitorConfig.ipv4.label', {
+      defaultMessage: 'IP protocols',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.ipv4.helpText', {
+      defaultMessage: 'Specifies which IP procols to use when pinging the remote host.',
+    }),
+    controlled: true,
+    dependencies: [ConfigKey.IPV6],
+    props: ({ field, setValue, dependencies }): EuiComboBoxProps<string> => {
+      const [ipv6] = dependencies;
+      const ipv4 = field?.value;
+      const values: string[] = [];
+      if (ipv4) {
+        values.push('IPv4');
+      }
+      if (ipv6) {
+        values.push('IPv6');
+      }
+      return {
+        options: [
+          {
+            label: 'IPv4',
+          },
+          {
+            label: 'IPv6',
+          },
+        ],
+        selectedOptions: values.map((version) => ({
+          label: version,
+        })),
+        onChange: (updatedValues: Array<EuiComboBoxOptionOption<string>>) => {
+          setValue(
+            ConfigKey.IPV4,
+            updatedValues.some((value) => value.label === 'IPv4')
+          );
+          setValue(
+            ConfigKey.IPV6,
+            updatedValues.some((value) => value.label === 'IPv6')
+          );
+        },
+        isDisabled: readOnly,
+      };
+    },
+  },
+  [ConfigKey.PROXY_HEADERS]: {
+    fieldKey: ConfigKey.PROXY_HEADERS,
+    component: HeaderField,
+    label: i18n.translate('xpack.synthetics.monitorConfig.proxyHeaders.label', {
+      defaultMessage: 'Proxy headers',
```

Comment:
Not very content related, but I'm wondering from the screenshots if it wouldn't be clearer to have the "+ Add expression", "+ Add header" and similar types of button below the setting descriptions and values already added. Just thinking in terms of layout. Maybe it'd look odd as well, not sure.